### DISCUSSION TITLE
Are you planning to continue?

### DISCUSSION THREAD

**Discussion by drakon660** [tye-v2/tye2#5] [General]
Is this project some sort of one time idea, or you have something more in your mind?

**Comment by razvangoga** [comment]
Hi!

It's a fork of a now obsoleted (and archived) oss project from Microsoft (Project Tye). It was superseded by Aspire.

I'm still using Tye in a couple of projects and wanted to keep it going (see readme and general v1 roadmap issue). I'm working on it as much as my real life allowes me.

Are you looking for something specifically from it?

**Comment by drakon660** [comment]
Maybe I was not precise enough, I know what a Tye was I was using it in company, but MS killed. I was searching through internet to find living fork and I found yours.I was curious about it because there is only one branch created few months ago. 
I know that you have a roadmap but it was nov 2023 and since then there was no commit. My curiosity was determined
 by engagement, I still think there is a place for such tool, so could help you what?

**Reply by razvangoga** [reply]
My plan in november was to hard fork it, merge all the improvement PRs that the MS folks ignored and get the cicd pipelines setup (publish to nuget as a new tool etc).

I only manged to merge all the PRs - got stuck with some strange errors while running the unit tests locally and did not have any more time for it (which was fine a for my use cases i could simply compile it locally).

Any help pushing it forward is much appreciated 👍🫡. 

As a general ideea - i'm open to pushing the tool forward **_for the local dev scenario_**, but **NOT** for the deployment one - i fundamentally think tye's deployment model is wrong, and aspire's is the way to go (with the intermediate manifest that can be later transpiled to which ever  deployment target you have).

**Reply by razvangoga** [reply]
But before starting on the ideeas outlined in the roadmap issue #2, i think the open points relating to the cicd from #1 must be closed.

And the first step there would be to have a working CI pipeline that will run the automated unit tests and which will serve as a stepping stone for publishing the tool a new dotnet tool (via nuget)

In parallel a decision must be reached about the name / naming / namespaces etc

**Comment by razvangoga** [comment]
@drakon660 i converted the issue you opened to a discussion - i would the issues only for  actual work items

**Reply by drakon660** [reply]
Regarding deployment using tye2 and aspire project, personally I think that are a lot better tools to do it in Kubernetes world. He is good article about aspire project https://event-driven.io/en/nay_to_aspire/. Beside that, I went through a few discussions and people are using it in production as orchestrator, so it is quite stable. I have experience using Azure DevOps for full product lifecycle, but GitHub is what it intrigues me more. My proposal is that I will fork your repo, check tests, try to fix it, then we could go with some POC  of which tool for pipeline we can take, so please check if you have something to push. Is this scenario ok with you?

**Reply by razvangoga** [reply]
## 1. Unit tests

i've updated all sample projects used as inputs for the unit tests to use net7 or 8 instead of 2.1, 3 5 and 6 ( just the test projects - the tye projects remain on net6). This caused some of them to fail. All failing tests are in this section 
<img width="527" alt="image" src="https://github.com/razvangoga/tye2/assets/1067494/d602c714-6c4e-4eb9-a132-b752cd3e158a">

All changes are checked in - you can fork away

I've opened a new issue #6 to track this - let's discuss there

## 2. CI Pipeline

i would also prefer to have the pipelines as Github Actions.

i see that any free github repo has 2000 minutes of action runtime / month - it should be enough.

I've opened a new issue #7 to track this - let's discuss there

-------------------

If you take a look at the unit tests as you proposed, i would try this week to create a an action pipeline. We could use it to validate any future pull requests (when all tests pass).

**Comment by drakon660** [comment]
CircleCi has 6000 minutes:)

**Comment by razvangoga** [comment]
I'm not familiar with it, but the 6k mins sound good.
I'll keep it mind if the GitHub Actions don't work out

**Comment by drakon660** [comment]
Lets close this discussion
